core/src/main/java/org/infinispan/container/DefaultDataContainer.java

```diff
@@ -350,6 +356,11 @@ public void evict(K key) {
       @Override
       public void onEntryEviction(Map<K, InternalCacheEntry<K, V>> evicted) {
          evictionManager.onEntryEviction(evicted);
+         List<CommandInterceptor> interceptors = interceptorChain.getInterceptorsWhichExtend(CacheMgmtInterceptor.class);
```

Comment:
This logic seems like a better fit in the EvictionManager instead of in the data container.